### PR TITLE
feat(search): added the search page

### PR DESCRIPTION
**Pull request by Shruti3004**
## Description

Added the Search page

## Changes

- Added the input container in the widgets. 
- Added the search mechanism.

## How to test

Visit the `/search` page

## Screenshots

Screenshot from 2021-06-28 11-58-18

Screenshot from 2021-06-28 11-58-24

### CODE REVIEW

**Comment by Shruti3004** [comment]
@GMishx how we get sub folders in this, we are getting the following data : 
```
filename: "$LogFile"
upload:{
    assignee: null
    description: "fossolgy"
    folderid: 1
    foldername: "Software Repository"
    hash:{
        md5: "954DD2992F91C61C91B9280E7986C16F"
        sha1: "3DF4C0B2804E691A29D7ABD00BD4FEFC3D21F36F"
        sha256: "B80A58791BB71D3DE62B595A5556ABB4EA55CF2734C167473B176B8DC047E6B6"
        size: 23953571
   }    
    id: 3,
    uploaddate: "2021-04-02 16:55:29.860327+05:30"
    uploadname: "fossology-master.zip"
}
uploadTreeId: 16134
```

**Comment by GMishx** [comment]
> @GMishx how we get sub folders in this, we are getting the following data :

Since current REST API does not allow browsing individual files, I made no sense to provide links to that. So, in the response all that's returned is the upload object (same as `/uploads/3`) and the `uploadTreeId` (which is the unique file ID in DB) and `filename` of the matched file.

**Comment by GMishx** [comment]
I am getting following error. Looks like auth token is missing from request.
```json
{"code":400,"message":"Authorization header is malformed or empty.","type":"ERROR"}
```

**Comment by Shruti3004** [comment]
> > @GMishx how we get sub folders in this, we are getting the following data :
> 
> Since current REST API does not allow browsing individual files, I made no sense to provide links to that. So, in the response all that's returned is the upload object (same as `/uploads/3`) and the `uploadTreeId` (which is the unique file ID in DB) and `filename` of the matched file.

Okay, So please checked the way I am representing the data on search page, shall I need to add or subtract anything from it.

File: src/pages/Search/index.js [hunk 1]
```diff
@@ -16,10 +16,175 @@
...
-  return <div>Search</div>;
+  const initialState = {
+    groupName: "",
+    searchType: "allfiles",
+    uploadId: "",
+    filename: "",
+    tag: "",
+    filesizemin: "",
+    filesizemax: "",
+    license: " ",
```
Comment (GMishx) [inline src/pages/Search/index.js:33]:
```suggestion
    license: "",
```

**Comment by Shruti3004** [comment]
> I am getting following error. Looks like auth token is missing from request.
> 
> ```json
> {"code":400,"message":"Authorization header is malformed or empty.","type":"ERROR"}
> ```

@GMishx, But its working fine for me

File: src/pages/Search/index.js [hunk 2]
```diff
@@ -16,10 +16,175 @@
...
+              name="filesizemax"
+              type="text"
+              id="search-file-size-max"
+              placeholder="in Bytes"
+              onChange={(e) => handleChange(e)}
+            >
+              File size is &le;
+            </InputContainer>
+            <div className="font-demi mt-4">
+              You may also choose one or more odivtional search filters (not
```
Comment (GMishx) [inline src/pages/Search/index.js:136]:
```suggestion
              You may also choose one or more optional search filters (not
```

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by GMishx** [comment]
> Okay, So please checked the way I am representing the data on search page, shall I need to add or subtract anything from it.

Current implementation is good enough. If you want, you can locally experiment by generating links from the response in format `http://localhost/repo/?mod=view-license&upload=<uploadid>&item=<uploadTreeId>`.

File: src/pages/Search/index.js (hunk 1 above)
Comment (Shruti3004) [reply src/pages/Search/index.js]: @GMishx Fixed

File: src/pages/Search/index.js (hunk 2 above)
Comment (Shruti3004) [reply src/pages/Search/index.js]: @GMishx Fixed

**Comment by Shruti3004** [comment]
> > Okay, So please checked the way I am representing the data on search page, shall I need to add or subtract anything from it.
> 
> Current implementation is good enough. If you want, you can locally experiment by generating links from the response in format `http://localhost/repo/?mod=view-license&upload=<uploadid>&item=<uploadTreeId>`.

Okay Sure

**Comment by GMishx** [comment]
Now it is working for me (with the `async`).

Can we have the help messages ("You can use '%' as a wild-card, For example: '%v3.war', or 'mypkg%.tar'.") or you are planning them for later?

**Comment by Shruti3004** [comment]
> Now it is working for me (with the `async`).
> 
> Can we have the help messages ("You can use '%' as a wild-card, For example: '%v3.war', or 'mypkg%.tar'.") or you are planning them for later?

Yeah Actually I am thinking to prepare a custom component for it like for all the popups to display all the helper messages and error messages

**Comment by Shruti3004** [comment]
> > Now it is working for me (with the `async`).
> > Can we have the help messages ("You can use '%' as a wild-card, For example: '%v3.war', or 'mypkg%.tar'.") or you are planning them for later?
> 
> Yeah Actually I am thinking to prepare a custom component for it like for all the popups to display all the helper messages and error messages.

**Comment by GMishx** [comment]
You can also use [tooltips](https://www.npmjs.com/package/react-tooltip) for help messages.

Do let me know once this PR is ready for merge.

**Comment by Shruti3004** [comment]
> You can also use [tooltips](https://www.npmjs.com/package/react-tooltip) for help messages.
> 
> Do let me know once this PR is ready for merge.

@GMishx Should i update the the tooltips in this pr itself?

**Comment by GMishx** [comment]
If it is not a blocker then it will be better to do it here for the search page. Or it can also be done in a separate PR for other components. Please create an issue if you want to do in separate PR.

**Comment by Shruti3004** [comment]
> If it is not a blocker then it will be better to do it here for the search page. Or it can also be done in a separate PR for other components. Please create an issue if you want to do in separate PR.

I think creating issue and doing it in a next pr will be a nicer option because then I will update the previous login code as well with that widgets.

**Comment by GMishx** [comment]
Sounds good to me :+1:

**Comment by Shruti3004** [comment]
> Sounds good to me

@GMishx created the issue #44

**Comment by Shruti3004** [comment]
@GMishx Does any other changes required in this pr?

**Review by GMishx — APPROVED** [review]

**Pull request merged**